test.js

```diff
@@ -827,3 +828,28 @@ test('Prints any other `--before-build` command error', (assert) => new Promise(
     'exits with whatever code the `--before-build` command returned'
   );
 }));
+
+test('debounce debounces', (assert) => new Promise((resolve) => {
+  assert.plan(1);
+
+  let timesRan = 0;
+  const countTimesRan = () => { timesRan += 1; };
+
+  const wait = 1;
+  const debouncedcountTimesRan = debounce(countTimesRan, wait);
+
+  debouncedcountTimesRan();
+  debouncedcountTimesRan();
+  debouncedcountTimesRan();
+  debouncedcountTimesRan();
+
+  const checkTimesRan = () => {
+    assert.is(timesRan, 1,
+      `only calls the passed function once when the returned function
+is called several times within the wait interval`
+    );
+    resolve();
+  };
+
+  setTimeout(checkTimesRan, wait + 1);
```

Comment:
How about bumping the difference between the wait and the check up? I’m a bit wary of race conditions at such small intervals – and that would mean flaky tests. How about 200ms? It’s slow for a unit test, but we don’t _really_ have to worry about build duration. 😄